job/src/main/java/org/jenkinsci/plugins/workflow/job/console/WorkflowMetadataConsoleFilter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.job.console;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import hudson.console.LineTransformationOutputStream;
+
+/**
+ * It transforms workflow metadata log messages through {@link WorkflowRunConsoleNote}.
+ */
+public class WorkflowMetadataConsoleFilter extends LineTransformationOutputStream {
+
+    private final OutputStream out;
+
+    public WorkflowMetadataConsoleFilter(OutputStream out) {
+        this.out = out;
+    }
+
+    @Override
+    protected void eol(byte[] b, int len) throws IOException {
+        new WorkflowRunConsoleNote().encodeTo(out);
```

Comment:
Do you need to create a new `WorkflowRunConsoleNote` for each line? Would one per `WorkflowMetadataConsoleFilter` not work?